### PR TITLE
fix(@aws-amplify/core): error throwing and tests

### PR DESCRIPTION
**Pull request by mzarnawski**
#### Description of changes
Yarn test was not working for me in core package. Fix in error throwing and assertions was needed

#### Description of how you validated changes
I ran tests in core scope and they passed. They are still failing for other packages but I will address them in a separate PR
BTW, these tests that I changed within this PR, did they run for all of you guys successfully ? I forked the repo a couple of days ago and was not able to run all tests.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes (in core scope)
- [x] Tests are changed (https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by evcodes** [comment]
Hey @mzarnawski,

Thank you for modifying these tests, we really appreciate your contribution. As for the failing tests you are running into, try `rm -rf node_modules && yarn install` to get fresh node modules. Then `yarn build && yarn test`. I was running into failing tests but those commands were able to do the trick for me. Finally, try using the latest version of node and see if that works for you.

You should be seeing all the tests pass, please let us know if that is not the case.

**Comment by mzarnawski** [comment]
Thanks @evcodes for you reply. I am running node 16.2.0, please let me know if this is too high or too low. I forked the repo a couple of days ago so I believe the modules are quite fresh, but I will reinstall them anyway to be sure. Currently I get like 3 failing tests in Auth module so I will examine these as well.

**Comment by evcodes** [comment]
I am using version 14.17.6 and all tests pass for my team. Try switching to that major version and see if it works.

**Comment by mzarnawski** [comment]
Hi again @evcodes, I downgraded to 14.17.6. and indeed test are passing but when I added "await" to expect(action()).rejects.toThrowError()
on line 79 of AWSCloudWatchProvider-test.ts
they started failing which means that "throw Error" indeed is not a proper way to throw error.

Long story short: with my corrections the tests finish successfully both for Node 14 and 16 hope that it helps you in the review.

**Comment by mzarnawski** [comment]
If you don't mind I will also address in a separate PR the 3 failing tests in Auth package, which will be necessary sooner or later if you plan to move to Node 16.

**Comment by mzarnawski** [comment]
@evcodes FYI, I took extra effort to check that this test starts to fail from Node v15.0.0 which is the next version after the one that you are using

File: packages/core/__tests__/Providers/AWSCloudWatchProvider-test.ts [hunk 1]
```diff
@@ -76,7 +76,8 @@ describe('AWSCloudWatchProvider', () => {
 				});
 			};
 
-			expect(action()).rejects.toThrowError();
+			expect.assertions(1);
+			await expect(action()).rejects.toThrowError();
```
Comment (evcodes) [inline packages/core/__tests__/Providers/AWSCloudWatchProvider-test.ts:80]: I would suggest to have it check that the error you are receiving in tests has the same message as the error being thrown from the package: 

`expect(action()).rejects.toThrowError(Error(NO_CREDS_ERROR_STRING));`

**Review by evcodes — CHANGES_REQUESTED** [review]
I made one suggestion but other than that, this looks fine to merge. Again thank you for going above and beyond!

File: packages/core/__tests__/Providers/AWSCloudWatchProvider-test.ts (hunk 1 above)
Comment (mzarnawski) [reply packages/core/__tests__/Providers/AWSCloudWatchProvider-test.ts]: Totally agree. Done

**Comment by mzarnawski** [comment]
Thank you very much, the pleasure is mine.

**Review by evcodes — APPROVED** [review]
LGTM, thank you for this contribution

**Comment by mzarnawski** [comment]
@evcodes you mean me ?
"The base branch restricts merging to authorized users"

**Comment by evcodes** [comment]
Sorry about that, I thought that with my approval you could have the pleasure of merging into Amplify 👍🏼

**Comment by mzarnawski** [comment]
Sooo nice to synchronize fork with my own changes :-)

**Pull request merged**